diff --git a/buildrump.sh/src/sys/rump/include/sys/bus.h b/buildrump.sh/src/sys/rump/include/sys/bus.h
index c7eedbcb5..bdc51f7f5 100644
--- a/buildrump.sh/src/sys/rump/include/sys/bus.h
+++ b/buildrump.sh/src/sys/rump/include/sys/bus.h
@@ -68,4 +68,12 @@ typedef struct {
 
 #include <sys/bus_proto.h>
 
+/*
+ * Convenience macros to correctly extract the upper and lower
+ * 32 bits of a bus_addr_t (which may be a 32-bit or 64-bit
+ * value).
+ */
+#define BUS_ADDR_HI32(a)        ((uint32_t) __SHIFTOUT(a, __BITS(32,63)))
+#define BUS_ADDR_LO32(a)        ((uint32_t) __SHIFTOUT(a, __BITS(0,31)))
+
 #endif /* _SYS_RUMP_BUS_H_ */
diff --git a/buildrump.sh/src/sys/dev/ic/ahcisata_core.c b/buildrump.sh/src/sys/dev/ic/ahcisata_core.c
index 8910f0a08..00a701094 100644
--- a/buildrump.sh/src/sys/dev/ic/ahcisata_core.c
+++ b/buildrump.sh/src/sys/dev/ic/ahcisata_core.c
@@ -36,6 +36,7 @@ __KERNEL_RCSID(0, "$NetBSD: ahcisata_core.c,v 1.100 2021/08/07 16:19:11 thorpej
 #include <sys/disklabel.h>
 #include <sys/proc.h>
 #include <sys/buf.h>
+#include <sys/bus.h>
 
 #include <dev/ata/atareg.h>
 #include <dev/ata/satavar.h>
diff --git a/buildrump.sh/src/sys/rump/dev/Makefile.rumpdevcomp b/buildrump.sh/src/sys/rump/dev/Makefile.rumpdevcomp
index 272304e3..44b607f6 100644
--- a/buildrump.sh/src/sys/rump/dev/Makefile.rumpdevcomp
+++ b/buildrump.sh/src/sys/rump/dev/Makefile.rumpdevcomp
@@ -8,6 +8,7 @@ RUMPDEVCOMP=	audio bpf cgd disk dm drvctl fss md netsmb		\
 RUMPUSBDEVS=	ubt ucom ugenhc ulpt umass usb
 
 RUMPPCIDEVS=	pci pci_if_iwn pci_if_pcn pci_if_wm
+RUMPPCIDEVS+=	pci_ahcisata
 RUMPPCIDEVS+=	pci_piixide
 RUMPPCIDEVS+=	pci_usbhc
 RUMPPCIDEVS+=	pci_virtio virtio_if_vioif virtio_ld virtio_viornd
diff --git a/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/AHCISATA.ioconf b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/AHCISATA.ioconf
new file mode 100644
index 00000000..2b7263bf
--- /dev/null
+++ b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/AHCISATA.ioconf
@@ -0,0 +1,19 @@
+ioconf ahcisata
+
+include "conf/files"
+include "dev/ata/files.ata"
+include "dev/pci/files.pci"
+include "rump/dev/files.rump"
+
+pseudo-root pci*
+
+# AHCI support
+ahcisata*	at pci? dev ? function ?
+atabus*		at ahcisata? channel ?
+ata_hl*		at atabus? drive ?
+
+# Disks
+wd*     	at ata_hl?
+
+# ATAPI support
+atapi*		at ahcisata?
diff --git a/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/Makefile b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/Makefile
new file mode 100644
index 00000000..1f6b870e
--- /dev/null
+++ b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/Makefile
@@ -0,0 +1,21 @@
+.PATH:	${.CURDIR}/../../../../dev/pci \
+	${.CURDIR}/../../../../dev/ata \
+	${.CURDIR}/../../../../dev/scsipi \
+	${.CURDIR}/../../../../dev/ic
+
+LIB=	rumpdev_ahcisata
+COMMENT=AHCI support with SATA && ATAPI
+
+IOCONF=	AHCISATA.ioconf
+
+SRCS+=	ahcisata_pci.c
+SRCS+=	sata_subr.c satafis_subr.c
+SRCS+=	ahcisata_core.c
+SRCS+=	atapi_wdc.c atapi_base.c
+SRCS+=	wd.c ata.c ata_wdc.c ata_subr.c
+SRCS+=	wdc.c
+
+SRCS+=	ahcisata_component.c
+
+.include <bsd.lib.mk>
+.include <bsd.klinks.mk>
diff --git a/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/ahcisata_component.c b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/ahcisata_component.c
new file mode 100644
index 00000000..0cc9f481
--- /dev/null
+++ b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/ahcisata_component.c
@@ -0,0 +1,40 @@
+#include <sys/param.h>
+#include <sys/conf.h>
+#include <sys/device.h>
+#include <sys/kmem.h>
+#include <sys/stat.h>
+
+#include "ioconf.c"
+
+#include <rump-sys/kern.h>
+#include <rump-sys/vfs.h>
+
+RUMP_COMPONENT(RUMP_COMPONENT_DEV)
+{
+	extern struct bdevsw wd_bdevsw;
+	extern struct cdevsw wd_cdevsw;
+	devmajor_t bmaj, cmaj;
+
+	config_init_component(cfdriver_ioconf_ahcisata,
+	    cfattach_ioconf_ahcisata, cfdata_ioconf_ahcisata);
+
+	bmaj = cmaj = -1;
+	FLAWLESSCALL(devsw_attach("wd", &wd_bdevsw, &bmaj, &wd_cdevsw, &cmaj));
+
+	FLAWLESSCALL(rump_vfs_makedevnodes(S_IFBLK, "/dev/wd0", 'a',
+	    bmaj, 0, 8));
+	FLAWLESSCALL(rump_vfs_makedevnodes(S_IFCHR, "/dev/rwd0", 'a',
+	    cmaj, 0, 8));
+	FLAWLESSCALL(rump_vfs_makedevnodes(S_IFBLK, "/dev/wd1", 'a',
+	    bmaj, 8, 8));
+	FLAWLESSCALL(rump_vfs_makedevnodes(S_IFCHR, "/dev/rwd1", 'a',
+	    cmaj, 8, 8));
+	FLAWLESSCALL(rump_vfs_makedevnodes(S_IFBLK, "/dev/wd2", 'a',
+	    bmaj, 16, 8));
+	FLAWLESSCALL(rump_vfs_makedevnodes(S_IFCHR, "/dev/rwd2", 'a',
+	    cmaj, 16, 8));
+	FLAWLESSCALL(rump_vfs_makedevnodes(S_IFBLK, "/dev/wd3", 'a',
+	    bmaj, 24, 8));
+	FLAWLESSCALL(rump_vfs_makedevnodes(S_IFCHR, "/dev/rwd3", 'a',
+	    cmaj, 24, 8));
+}
diff --git a/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/opt_wd.h b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/opt_wd.h
new file mode 100644
index 00000000..e69de29b
--- /dev/null
+++ b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/opt_wd.h
@@ -0,0 +1 @@
+
diff --git a/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/ataraid.h b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/ataraid.h
new file mode 100644
index 00000000..e69de29b
--- /dev/null
+++ b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/ataraid.h
@@ -0,0 +1 @@
+
diff --git a/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/opt_ata.h b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/opt_ata.h
new file mode 100644
index 00000000..e69de29b
--- /dev/null
+++ b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/opt_ata.h
@@ -0,0 +1 @@
+
diff --git a/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/opt_wdc.h b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/opt_wdc.h
new file mode 100644
index 00000000..ad9cf43a
--- /dev/null
+++ b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/opt_wdc.h
@@ -0,0 +1,3 @@
+#define NWD	1
+#define NSATA	1
+#define WDC_NO_IDS
diff --git a/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/sata.h b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/sata.h
new file mode 100644
index 00000000..e69de29b
--- /dev/null
+++ b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/sata.h
@@ -0,0 +1 @@
+
diff --git a/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/sata_pmp.h b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/sata_pmp.h
new file mode 100644
index 00000000..e69de29b
--- /dev/null
+++ b/buildrump.sh/src/sys/rump/dev/lib/libpci_ahcisata/sata_pmp.h
@@ -0,0 +1 @@
+
