diff --git a/buildrump.sh/buildrump.sh b/buildrump.sh/buildrump.sh
index 41afb5cd..dab869cd 100755
--- a/buildrump.sh/buildrump.sh
+++ b/buildrump.sh/buildrump.sh
@@ -1047,6 +1047,9 @@ evaltoolchain ()
 		;;
 	*-gnu*)
 		RUMPKERN_UNDEF='-U__GNU__'
+		appendvar EXTRA_CWARNFLAGS -Wno-error
+		appendvar EXTRA_CWARNFLAGS -Wno-format
+		appendvar EXTRA_CWARNFLAGS -Wno-format-security
 		cppdefines _BIG_ENDIAN \
 		    && appendvar RUMPKERN_UNDEF -U_BIG_ENDIAN
 		cppdefines _LITTLE_ENDIAN \
