
Status: Merged in buildrump.sh and src-netbsd

Index: rumpkernel/buildrump.sh/buildrump.sh
===================================================================
--- rumpkernel.orig/buildrump.sh/buildrump.sh
+++ rumpkernel/buildrump.sh/buildrump.sh
@@ -993,6 +993,13 @@ evaltoolchain ()
 		cppdefines _LITTLE_ENDIAN \
 		    && appendvar RUMPKERN_UNDEF -U_LITTLE_ENDIAN
 		;;
+	*-gnu*)
+		RUMPKERN_UNDEF='-U__GNU__'
+		cppdefines _BIG_ENDIAN \
+		    && appendvar RUMPKERN_UNDEF -U_BIG_ENDIAN
+		cppdefines _LITTLE_ENDIAN \
+		    && appendvar RUMPKERN_UNDEF -U_LITTLE_ENDIAN
+		;;
 	*-dragonflybsd)
 		RUMPKERN_UNDEF='-U__DragonFly__'
 		;;
@@ -1064,6 +1071,10 @@ evalplatform ()
 		doesitbuild '#include <linux/if_tun.h>' -c && RUMP_VIRTIF=yes
 		cppdefines '__ANDROID__' || HIJACK=true
 		;;
+	*-gnu*)
+		EXTRA_RUMPCOMMON='-ldl'
+		EXTRA_RUMPCLIENT='-lpthread'
+		;;
 	*-openbsd*)
 		EXTRA_RUMPCLIENT='-lpthread'
 		;;
Index: rumpkernel/buildrump.sh/src/lib/librumpuser/rumpuser_port.h
===================================================================
--- rumpkernel.orig/buildrump.sh/src/lib/librumpuser/rumpuser_port.h
+++ rumpkernel/buildrump.sh/src/lib/librumpuser/rumpuser_port.h
@@ -63,7 +63,7 @@
 #include "rumpuser_config.h"
 #endif
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__GNU__) || defined(__GLIBC__)
 #define _GNU_SOURCE
 #endif
 
Index: rumpkernel/buildrump.sh/src/lib/librumpuser/rumpuser_sp.c
===================================================================
--- rumpkernel.orig/buildrump.sh/src/lib/librumpuser/rumpuser_sp.c
+++ rumpkernel/buildrump.sh/src/lib/librumpuser/rumpuser_sp.c
@@ -90,7 +90,7 @@ static char banner[MAXBANNER];
 
 
 /* how to use atomic ops on Linux? */
-#if defined(__linux__) || defined(__APPLE__) || defined(__CYGWIN__) || defined(__OpenBSD__)
+#if defined(__linux__) || defined(__APPLE__) || defined(__CYGWIN__) || defined(__OpenBSD__) || defined(__GNU__) || defined(__GLIBC__)
 static pthread_mutex_t discomtx = PTHREAD_MUTEX_INITIALIZER;
 
 static void
Index: rumpkernel/buildrump.sh/src/sys/sys/syscallargs.h
===================================================================
--- rumpkernel.orig/buildrump.sh/src/sys/sys/syscallargs.h
+++ rumpkernel/buildrump.sh/src/sys/sys/syscallargs.h
@@ -10,8 +10,8 @@
 #ifndef _SYS_SYSCALLARGS_H_
 #define	_SYS_SYSCALLARGS_H_
 
-#include <sys/mount.h>
 #ifndef RUMP_CLIENT
+#include <sys/mount.h>
 #include <sys/sched.h>
 #endif
 #include <sys/socket.h>
