Index: rumpkernel/buildrump.sh/src/sys/rump/include/rump/rump_syscalls_compat.h
===================================================================
--- rumpkernel.orig/buildrump.sh/src/sys/rump/include/rump/rump_syscalls_compat.h
+++ rumpkernel/buildrump.sh/src/sys/rump/include/rump/rump_syscalls_compat.h
@@ -37,6 +37,7 @@
 #include <sys/types.h> /* typedefs */
 #include <sys/select.h> /* typedefs */
 #include <sys/socket.h> /* typedefs */
+#include <sys/wait.h> /* typedefs */
 
 #include <signal.h> /* typedefs */
 #endif
