diff --git a/pci-userspace/src-gnu/pci_user-gnu.c b/pci-userspace/src-gnu/pci_user-gnu.c
index ef48ab2a..86d8c2af 100644
--- a/pci-userspace/src-gnu/pci_user-gnu.c
+++ b/pci-userspace/src-gnu/pci_user-gnu.c
@@ -170,7 +170,8 @@ rumpcomp_pci_map(unsigned long addr, unsigned long len)
 found:
 
 	pci_device_map_range(pci_devices[i], addr, len, 1, &ret);
-	return pci_devices[i]->regions[residx].memory;
+	pci_devices[i]->regions[residx].memory = ret;
+	return ret;
 }
 
 int
