Index: rumpkernel/pci-userspace/src-gnu/Makefile.inc
===================================================================
--- rumpkernel.orig/pci-userspace/src-gnu/Makefile.inc
+++ rumpkernel/pci-userspace/src-gnu/Makefile.inc
@@ -4,8 +4,8 @@ PCIDIR:=	${.PARSEDIR}
 .PATH:		${PCIDIR}
 
 RUMPCOMP_USER_SRCS=	pci_user-gnu.c mach_debugUser.c
-RUMPCOMP_USER_CPPFLAGS+=-I${PCIDIR}
-RUMPCOMP_CPPFLAGS+=	-I${PCIDIR}
+RUMPCOMP_USER_CPPFLAGS+=-I${PCIDIR} -I${DESTDIR}/usr/include
+RUMPCOMP_CPPFLAGS+=	-I${PCIDIR} -I${DESTDIR}/usr/include
 CPPFLAGS+=		-I${PCIDIR}
 LDFLAGS+= -lmachuser -Wl,--no-as-needed -lpciaccess -Wl,--as-needed
 
