Index: rumpkernel/buildrump.sh/src/build.sh
===================================================================
--- rumpkernel.orig/buildrump.sh/src/build.sh
+++ rumpkernel/buildrump.sh/src/build.sh
@@ -403,7 +403,7 @@ set_HOST_SH()
 	# specifying HOST_SH in the environment.
 	#
 	[ -z "${HOST_SH}" ] && HOST_SH="$(
-		(ps -p $$ -o comm | sed -ne "2s/[ ${tab}]*\$//p") 2>/dev/null )"
+		(ps -p $$ -o comm | sed -n -e "s/^[ ${tab}]*//" -e "2s/[ ${tab}]*\$//p") 2>/dev/null )"
 
 	# If nothing above worked, use "sh".  We will later find the
 	# first directory in the PATH that has a "sh" program.
