Index: rumpkernel/buildrump.sh/src/tools/Makefile
===================================================================
--- rumpkernel.orig/buildrump.sh/src/tools/Makefile	2015-09-07 22:40:48.353751980 +0200
+++ rumpkernel/buildrump.sh/src/tools/Makefile	2016-03-06 00:20:38.921308141 +0100
@@ -102,7 +102,7 @@
 		makewhatis mtree nbperf .WAIT uudecode
 .endif
 
-SUBDIR+= cat rpcgen join lorder m4 mkdep tsort .WAIT yacc .WAIT awk .WAIT lex
+SUBDIR+= cat join lorder m4 mkdep tsort .WAIT yacc .WAIT awk .WAIT lex
 
 .if ${TOOLS_BUILDRUMP} == "no"
 SUBDIR += .WAIT texinfo \
