diff --git a/buildrump.sh/src/lib/librumpuser/rumpuser_port.h b/buildrump.sh/src/lib/librumpuser/rumpuser_port.h
index 5c5cc8f83..467ac86e8 100644
--- a/buildrump.sh/src/lib/librumpuser/rumpuser_port.h
+++ b/buildrump.sh/src/lib/librumpuser/rumpuser_port.h
@@ -19,7 +19,6 @@
 #define HAVE_CLOCK_GETTIME 1
 #define HAVE_CLOCK_NANOSLEEP 1
 #define HAVE_DLINFO 1
-#define HAVE_FSYNC_RANGE 1
 #define HAVE_GETENV_R 1
 #define HAVE_GETPROGNAME 1
 #define HAVE_GETSUBOPT 1
@@ -29,6 +28,7 @@
 #define HAVE_PATHS_H 1
 #define HAVE_POSIX_MEMALIGN 1
 #define HAVE_PTHREAD_SETNAME3 1
+#define HAVE_REGISTER_T 1
 #define HAVE_SETPROGNAME 1
 #define HAVE_STDINT_H 1
 #define HAVE_STDLIB_H 1
